cos-to-sql/app.mjs

```diff
@@ -0,0 +1,206 @@
+const __dirname = path.dirname(__filename); // get the name of the directory
+
+// Libraries needed to materialize the authentication to COS and to SecretsManager
+// and to read service credentials from SecretsManager
+import SecretsManager from "@ibm-cloud/secrets-manager/secrets-manager/v2.js";
+import { ContainerAuthenticator } from "ibm-cloud-sdk-core";
+
+// Imports to handle data base related operations
+import { addUser, closeDBConnection, deleteUsers, getPgClient, listUsers } from "./utils/db.mjs";
+import { getObjectContent } from "./utils/cos.mjs";
+import { convertCsvToDataStruct } from "./utils/utils.mjs";
+
+console.info("Starting the app ...");
+
+const requiredEnvVars = [
+  "COS_REGION",
+  "COS_TRUSTED_PROFILE_NAME",
+  "SM_TRUSTED_PROFILE_NAME",
+  "SM_SERVICE_URL",
+  "SM_PG_SECRET_ID",
+];
+
+requiredEnvVars.forEach((envVarName) => {
+  if (!process.env[envVarName]) {
+    console.log(`Failed to start app. Missing '${envVarName}' environment variable`);
+    process.exit(1);
+  }
+});
+
+//
+// Initialize COS
+const cosRegion = process.env.COS_REGION;
+const cosTrustedProfileName = process.env.COS_TRUSTED_PROFILE_NAME;
+let cosAuthenticator;
+if (cosTrustedProfileName) {
+  // create an authenticator to access the COS instance based on a trusted profile
+  cosAuthenticator = new ContainerAuthenticator({
+    iamProfileName: cosTrustedProfileName,
+  });
+}
+
+//
+// Initialize Secrets Manager SDK
+const smTrustedProfileName = process.env.SM_TRUSTED_PROFILE_NAME;
+const smServiceURL = process.env.SM_SERVICE_URL;
+const smPgSecretId = process.env.SM_PG_SECRET_ID;
+let secretsManager;
+if (smTrustedProfileName && smServiceURL) {
+  // create an authenticator to access the SecretsManager instance based on a trusted profile
+  const smAuthenticator = new ContainerAuthenticator({
+    iamProfileName: smTrustedProfileName,
+  });
+  // Create an instance of the SDK by providing an authentication mechanism and your Secrets Manager instance URL
+  secretsManager = new SecretsManager({
+    authenticator: smAuthenticator,
+    serviceUrl: smServiceURL,
+  });
+}
+
+const app = express();
+app.use(express.json());
+app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')))
+
+// use router to bundle all routes to /
+const router = express.Router();
+app.use("/", router);
+
+
+//
+// Default http endpoint, which prints the list of all users in the database
+router.get("/", async (req, res) => {
+  console.log(`handling / for '${req.url}'`);
+  const pgClient = await getPgClient(secretsManager, smPgSecretId);
+  const allUsers = await listUsers(pgClient);
+  res.writeHead(200, { "Content-Type": "application/json" });
+  res.end(JSON.stringify({ users: allUsers.rows }));
+});
+
+//
+// Readiness endpoint
+router.get("/readiness", (req, res) => {
+  console.log(`handling /readiness`);
+  if (!existsSync("/var/run/secrets/codeengine.cloud.ibm.com/compute-resource-token/token")) {
```

Comment:
Import from `fs/promises`, then use `await` here to be non-blocking.

Reply:
Thanks for spotting this! Especially, in our samples we should make sure to not do such mistakes that one would regret if this app is used at scale